recipe-box-frontend/src/main/frontend/src/components/RecipesList/RecipesGrid.tsx

```diff
@@ -25,10 +27,14 @@ export default function RecipesGrid({
         const controller = new AbortController();
         const signal = controller.signal;
 
+        const storedUserId = sessionStorage.getItem("userId");
+        const userId = storedUserId !== null ? parseInt(storedUserId, 10) : undefined;
```

Comment:
You can have these in a Utils/user.ts  folder/file as these seem to not depend on what is in the files and are used in multiple places. You can transform these into functions (e.g. 
`export const storedUserId = () => sessionStorage.getItem("userId"`
)